app/assets/javascripts/media_player_wrapper/mejs4_plugin_playlist_items.es6

```diff
@@ -270,58 +287,101 @@ Object.assign(MediaElementPlayer.prototype, {
       const plo = t.playlistItemsObj;
       const currentTime = plo.player.getCurrentTime();
 
-      // Current time is within range of playlist item's start / end times
-      if (
-        currentTime >= plo.startEndTimes.start &&
-        currentTime < plo.startEndTimes.end
-      ) {
+      // The player currentTime is less than item end time, so revert to regular behavior
+      // of playing through time ranges
+      if (currentTime < plo.startEndTimes.end) {
+        plo.seekPastEnd = false;
+      }
+
+      // Do nothing, current time is within playlist item start / end range
+      if (plo.isCurrentTimeInRange(currentTime)) {
         return;
       }
+
       // Playlist item's end time is reached
-      if (plo.itemEnded() && !plo.refreshInProgress) {
+      if (
+        plo.isItemEnded(currentTime) &&
+        !plo.refreshInProgress &&
+        !plo.seekPastEnd
+      ) {
         plo.handleRangeEndTimeReached();
         return;
       }
     },
 
     /**
      * Handle user's manual seeking
+     * @function handleUserSeeking
+     * @param {MouseEvent} e Event emitted from the 'seeking'
      * @return {void}
      */
     handleUserSeeking(e) {
+      const plo = this.playlistItemsObj;
+      const currentTime = plo.player.getCurrentTime();
+
       // Always turn off Autoplay when user starts seeking around
-      this.playlistItemsObj.turnOffAutoplay();
+      plo.turnOffAutoplay();
+      // User seeked past the item end point
+      if (plo.isItemEnded(currentTime)) {
+        plo.seekPastEnd = true;
+      }
     },
 
-    refreshInProgress: false,
-
     /**
-     * Turn off Autoplay (auto advancing to next item)
-     * @return {void}
+     * Does the DOM reflect that Autoplay is 'On'? (Note: DOM checkbox value is the source of truth)
+     * @function isAutoplay
+     * @return {Boolean} [description]
      */
-    turnOffAutoplay() {
-      $('input[name="autoadvance"]')
-        .prop('checked', false)
-        .change();
-    },
-
     isAutoplay() {
       return !$('input[name="autoadvance"]')
         .parent('.toggle')
         .hasClass('off');
     },
 
+    /**
+     * Helper function: is the player's current time within the playlist items start / end time range?
+     * Note the 'threshold' helper variable, and see description below for why it's needed.
+     * @function isCurrentTimeInRange
+     * @param  {number}  currentTime Current time of Mediaelement player
+     * @return {Boolean}
+     */
+    isCurrentTimeInRange(currentTime) {
+      const currentTimeAdjusted = currentTime + this.threshold;
+      const startEndTimes = this.startEndTimes;
+      return (
+        currentTimeAdjusted >= startEndTimes.start &&
```

Comment:
This could probably be ```currentTime >= startEndTimes.start```. I don't think we need to adjust for the start times.

Reply:
Sure.  I'll adjust both.